File: solutions/938/c/solution.c
// 938. Range Sum of BST
#include <stddef.h>

int rangeSumBST(struct TreeNode* root, int L, int R)
{
	if (root == NULL)
	{
		return 0;
	}

	int value = 0;

	if (root->val >= L && root->val <= R)
	{
		value += root->val;
	}

	return value + rangeSumBST(root->left, L, R) + rangeSumBST(root->right, L, R);
}

File: solutions/938/java/Solution.java
// 938. Range Sum of BST
class Solution
{
	public int rangeSumBST(TreeNode root, int L, int R)
	{
		if (root == null)
		{
			return 0;
		}

		int sum = 0;

		if (root.val >= L && root.val <= R)
		{
			sum = root.val;
		}

		return sum + rangeSumBST(root.left, L, R) + rangeSumBST(root.right, L, R);
	}
}